src/bitcoin-cli.cpp

```diff
@@ -239,16 +240,17 @@ class GetinfoRequestHandler: public BaseRequestHandler
         result.push_back(JSONRPCRequestObj("getnetworkinfo", NullUniValue, ID_NETWORKINFO));
         result.push_back(JSONRPCRequestObj("getblockchaininfo", NullUniValue, ID_BLOCKCHAININFO));
         result.push_back(JSONRPCRequestObj("getwalletinfo", NullUniValue, ID_WALLETINFO));
+        result.push_back(JSONRPCRequestObj("getbalances", NullUniValue, ID_BALANCES));
         return result;
     }
 
     /** Collect values from the batch and form a simulated `getinfo` reply. */
     UniValue ProcessReply(const UniValue &batch_in) override
     {
         UniValue result(UniValue::VOBJ);
-        std::vector<UniValue> batch = JSONRPCProcessBatchReply(batch_in, 3);
-        // Errors in getnetworkinfo() and getblockchaininfo() are fatal, pass them on
-        // getwalletinfo() is allowed to fail in case there is no wallet.
+        std::vector<UniValue> batch = JSONRPCProcessBatchReply(batch_in, batch_in.size());
```

Comment:
Could change `JSONRPCProcessBatchReply` to just take `batch_in` (drop 2nd arg)?

Reply:
Thanks for reviewing. Was just looking at doing that in the follow-up.

Reply:
```diff
src/bitcoin-cli.cpp
     UniValue ProcessReply(const UniValue &batch_in) override
     {
         UniValue result(UniValue::VOBJ);
-        std::vector<UniValue> batch = JSONRPCProcessBatchReply(batch_in, batch_in.size());
+        std::vector<UniValue> batch = JSONRPCProcessBatchReply(batch_in);

src/rpc/request.cpp
@@ -130,11 +130,12 @@ void DeleteAuthCookie()
     }
 }

-std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue &in, size_t num)
+std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue &in)
 {
     if (!in.isArray()) {
         throw std::runtime_error("Batch must be an array");
     }
+    auto num{in.size()};
     std::vector<UniValue> batch(num);
     for (size_t i=0; i<in.size(); ++i) {
         const UniValue &rec = in[i];

src/rpc/request.h
 /** Parse JSON-RPC batch reply into a vector */
-std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue &in, size_t num);
+std::vector<UniValue> JSONRPCProcessBatchReply(const UniValue &in);
```

Reply:
TBH I'd change it now.